src/main/java/CommandParser.java

```diff
@@ -0,0 +1,84 @@
+package main.java;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CommandParser {
+    private String input;
+    private String type;
+
+    public CommandParser(String input) {
+        this.input = input;
+        int firstSpaceIndex = input.indexOf(" ");
+        if (firstSpaceIndex != -1) {
+            this.type = input.substring(0, firstSpaceIndex);
+        } else {
+            this.type = input;
+        }
+    }
+
+    public String getType() {
+        return this.type;
+    }
+
+    public String[] parseInput() throws ChatException {
+        switch (this.type) {
+            case "todo":
+                String regex1 = "^(\\w+)(\\s)(.*)";
+                Pattern pattern1 = Pattern.compile(regex1);
+                Matcher matcher1 = pattern1.matcher(input);
+                if (matcher1.find()) {
+                    String description = matcher1.group(3);
+                    return new String[]{description};
+                } else {
+                    throw new ChatException("Input does not match expected format: todo <task>");
+                }
+            case "deadline":
+                String regex2 = "^(\\w+) (.+?)\\/by (.+)$";
+                Pattern pattern2 = Pattern.compile(regex2);
+                Matcher matcher2 = pattern2.matcher(input);
+                if (matcher2.find()) {
+                    String description = matcher2.group(2);
+                    String toTime = matcher2.group(3);
+                    return new String[]{description, toTime};
+                } else {
+                    throw new ChatException("Input does not match expected format: deadline <task> /by <time>");
+                }
+            case "event":
+                String regex3 = "^(\\w+) (.+?) \\/from (.+?) \\/to (.+)$";
+                Pattern pattern3 = Pattern.compile(regex3);
+                Matcher matcher3 = pattern3.matcher(input);
+                if (matcher3.find()) {
+                    String description = matcher3.group(2);
+                    String fromTime = matcher3.group(3);
+                    String toTime = matcher3.group(4);
+                    return new String[]{description, fromTime, toTime};
+                } else {
+                    throw new ChatException("Input does not match expected format: event <task> /from <time> /to <time>");
+                }
+            case "mark":
+            case "unmark":
```

Comment:
As stated in the Java coding standard, the explicit `//Fallthrough` comment should be included whenever there is a `case` statement without a break statement.